### DISCUSSION TITLE
Saving MQTT messages to TimescaleDb

### DISCUSSION THREAD

**Discussion by pbwur** [vernemq/vernemq#2281] [General]
Hi,

I'm working on a POC where I would like to combine VerneMq as the MQTT broker and TimescaleDb/Postgres as the Timeseries database. 
At first I was thinking of using e.g. Telegraf for subscribing to VerneMq and saving the messages to the database. But because the (json) messages can have all sorts of structure, that would mean I have to do a lot of tricky transformations.
Now I'm looking at using node-red in the middle. But I'm concerned about performance and throughput.

Is there anyone who has done this before and has advice?

**Comment by ioolkos** [comment]
Apache NiFi is another option, especially when you have to do transformations within the flow. OTOH NodeRed is easy to set up (I think), so why not test it against your load scenario.
A native VerneMQ plugin would be a comparatively low effort but might not be the best approach when you need generic message transformations. You'd also have to configure the topics to export, in this case internally (via vernemq.conf), instead of external MQTT consumers.
